pkg/cmd/server/api/v1/types.go

```diff
@@ -138,6 +145,11 @@ type MasterConfig struct {
 
 	// NetworkConfig to be passed to the compiled in network plugin
 	NetworkConfig NetworkConfig `json:"networkConfig"`
+
+	// Enable or disable OpenShift components.
+	// TODO: find out how to write this field only if any of its field contains
+	// non-default value.
+	FeatureConfig FeatureConfig `json:"-"`
```

Comment:
Let's plan on separate fields.  That would allow this to simply be a `[]string` of disabled features that is `omitempty` with a set of helpers for specific fields.  This would also ensure that we never lost values while roundtripping through older binaries.